components/raftstore/src/store/fsm/peer.rs

```diff
@@ -410,20 +415,15 @@ where
         false
     }
 
-    fn build(&mut self, metric: &mut RaftMetrics) -> Option<RaftCommand<E::Snapshot>> {
+    fn build(
+        &mut self,
+        metric: &mut RaftMetrics,
+    ) -> Option<(RaftCmdRequest, Callback<E::Snapshot>)> {
         if let Some(req) = self.request.take() {
             self.batch_req_size = 0;
             if self.callbacks.len() == 1 {
-                let (mut cb, _, send_time) = self.callbacks.pop().unwrap();
-                if let Callback::Write { request_times, .. } = &mut cb {
-                    if metric.waterfall_metrics {
-                        metric
-                            .batch_wait
-                            .observe(send_time.saturating_elapsed_secs());
-                    }
-                    *request_times = smallvec![send_time];
```

Comment:
Why delete?

Reply:
Some cmd may not be batched so the `batch_wait` metric is moved to `propose_raft_command`.